In [10]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
import statistics

In [11]:
df = pd.read_csv("C:\\Users\\kalta\\OneDrive\\DSA\\Documents\\Downloads\\archive (3)\\ecommerce_customer_data_large.csv")

In [12]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 250000 entries, 0 to 249999
Data columns (total 13 columns):
 #   Column                 Non-Null Count   Dtype  
---  ------                 --------------   -----  
 0   Customer ID            250000 non-null  int64  
 1   Purchase Date          250000 non-null  object 
 2   Product Category       250000 non-null  object 
 3   Product Price          250000 non-null  int64  
 4   Quantity               250000 non-null  int64  
 5   Total Purchase Amount  250000 non-null  int64  
 6   Payment Method         250000 non-null  object 
 7   Customer Age           250000 non-null  int64  
 8   Returns                202618 non-null  float64
 9   Customer Name          250000 non-null  object 
 10  Age                    250000 non-null  int64  
 11  Gender                 250000 non-null  object 
 12  Churn                  250000 non-null  int64  
dtypes: float64(1), int64(7), object(5)
memory usage: 24.8+ MB


Returns have  47382 missing values

In [13]:
print(df.isnull().sum())

Customer ID                  0
Purchase Date                0
Product Category             0
Product Price                0
Quantity                     0
Total Purchase Amount        0
Payment Method               0
Customer Age                 0
Returns                  47382
Customer Name                0
Age                          0
Gender                       0
Churn                        0
dtype: int64


# Handling Missing Values 
The 'Returns' column is a binary indicator (0 for no return, 1 for a return).

The missing values should be interpreted as 0, meaning "no return."

In [14]:
updated_df = df
df['Returns'] = df['Returns'].fillna(0)
updated_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 250000 entries, 0 to 249999
Data columns (total 13 columns):
 #   Column                 Non-Null Count   Dtype  
---  ------                 --------------   -----  
 0   Customer ID            250000 non-null  int64  
 1   Purchase Date          250000 non-null  object 
 2   Product Category       250000 non-null  object 
 3   Product Price          250000 non-null  int64  
 4   Quantity               250000 non-null  int64  
 5   Total Purchase Amount  250000 non-null  int64  
 6   Payment Method         250000 non-null  object 
 7   Customer Age           250000 non-null  int64  
 8   Returns                250000 non-null  float64
 9   Customer Name          250000 non-null  object 
 10  Age                    250000 non-null  int64  
 11  Gender                 250000 non-null  object 
 12  Churn                  250000 non-null  int64  
dtypes: float64(1), int64(7), object(5)
memory usage: 24.8+ MB


In [15]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 250000 entries, 0 to 249999
Data columns (total 13 columns):
 #   Column                 Non-Null Count   Dtype  
---  ------                 --------------   -----  
 0   Customer ID            250000 non-null  int64  
 1   Purchase Date          250000 non-null  object 
 2   Product Category       250000 non-null  object 
 3   Product Price          250000 non-null  int64  
 4   Quantity               250000 non-null  int64  
 5   Total Purchase Amount  250000 non-null  int64  
 6   Payment Method         250000 non-null  object 
 7   Customer Age           250000 non-null  int64  
 8   Returns                250000 non-null  float64
 9   Customer Name          250000 non-null  object 
 10  Age                    250000 non-null  int64  
 11  Gender                 250000 non-null  object 
 12  Churn                  250000 non-null  int64  
dtypes: float64(1), int64(7), object(5)
memory usage: 24.8+ MB


In [16]:
df.head()

Customer ID        Purchase Date Product Category  Product Price  Quantity  \
0        44605  2023-05-03 21:30:02             Home            177         1   
1        44605  2021-05-16 13:57:44      Electronics            174         3   
2        44605  2020-07-13 06:16:57            Books            413         1   
3        44605  2023-01-17 13:14:36      Electronics            396         3   
4        44605  2021-05-01 11:29:27            Books            259         4   

   Total Purchase Amount Payment Method  Customer Age  Returns Customer Name  \
0                   2427         PayPal            31      1.0   John Rivera   
1                   2448         PayPal            31      1.0   John Rivera   
2                   2345    Credit Card            31      1.0   John Rivera   
3                    937           Cash            31      0.0   John Rivera   
4                   2598         PayPal            31      1.0   John Rivera   

   Age  Gender  Churn  
0   31  Female      0  
1   31  Female      0  
2   31  Female      0  
3   31  Female      0  
4   31  Female      0

In [17]:
type(df)

pandas.core.frame.DataFrame

In [19]:
df.head(8)

Customer ID        Purchase Date Product Category  Product Price  Quantity  \
0        44605  2023-05-03 21:30:02             Home            177         1   
1        44605  2021-05-16 13:57:44      Electronics            174         3   
2        44605  2020-07-13 06:16:57            Books            413         1   
3        44605  2023-01-17 13:14:36      Electronics            396         3   
4        44605  2021-05-01 11:29:27            Books            259         4   
5        13738  2022-08-25 06:48:33             Home            191         3   
6        13738  2023-07-25 05:17:24      Electronics            205         1   
7        13738  2023-02-05 19:31:48            Books            370         5   

   Total Purchase Amount Payment Method  Customer Age  Returns  \
0                   2427         PayPal            31      1.0   
1                   2448         PayPal            31      1.0   
2                   2345    Credit Card            31      1.0   
3                    937           Cash            31      0.0   
4                   2598         PayPal            31      1.0   
5                   3722    Credit Card            27      1.0   
6                   2773    Credit Card            27      0.0   
7                   1486           Cash            27      1.0   

    Customer Name  Age  Gender  Churn  
0     John Rivera   31  Female      0  
1     John Rivera   31  Female      0  
2     John Rivera   31  Female      0  
3     John Rivera   31  Female      0  
4     John Rivera   31  Female      0  
5  Lauren Johnson   27  Female      0  
6  Lauren Johnson   27  Female      0  
7  Lauren Johnson   27  Female      0

In [20]:
df.duplicated().sum()

np.int64(0)

In [24]:
# Check the counts of churned vs. retained customers
churn_counts = df['Churn'].value_counts()
print(churn_counts)

Churn
0    199870
1     50130
Name: count, dtype: int64


In [25]:
 # Calculate the churn rate
total_customers = churn_counts.sum()
churned_customers = churn_counts.get(1, 0) # Use .get() to avoid errors if there are no churned customers
churn_rate = (churned_customers / total_customers) * 100
print(f"\nOverall Churn Rate: {churn_rate:.2f}%")


Overall Churn Rate: 20.05%


* Key Findings from Initial EDA
* No Duplicates Found: There are no duplicate rows in the dataset. 
* Overall Churn Rate: The initial analysis reveals a significant overall churn rate of 20.05%. This is a critical business finding that validates the need for this project. My future analysis will now focus on identifying the factors and customer segments driving this churn.



In [27]:
import pandas as pd
from sqlalchemy import create_engine

# Use the same credentials as your mysql.connector code
# The format is: 'mysql+mysqlconnector://user:password@host:port/database_name'
try:
    engine = create_engine('mysql+mysqlconnector://root:root@127.0.0.1:3306/ecommerce')
    
    # Assuming your cleaned DataFrame is named 'df_cleaned'
    # This will create a new table named 'ecommerce_data' in your MySQL database
    df.to_sql('ecommerce_data', con=engine, if_exists='replace', index=False)
    
    print("Data successfully loaded into the MySQL database.")

except Exception as e:
    print(f"An error occurred: {e}")

Data successfully loaded into the MySQL database.
